minutes/2018-10-12-node-interactive.md

```diff
@@ -0,0 +1,199 @@
+
+### Daniel Stockman (@evocateur)
+
+* works at Zillow.
+* maintains Lerna.
+
+### Erick Wendel (@erickwendel)
+
+* Microsoft MVP.
+* curious about tooling (would like to contribute more).
+
+### Ruy Adorno (@ruyadorno):
+
+* contributor to yeoman.
+
+### Michel Lopez (@michellopez):
+
+* senior program manager at Microsoft.
+* interested in making sure Node.js/npm are stable on Windows.
+* excited that Windows and Linux build times are now equivilent.
+
+### Bryan English (@bengl)
+
+* engineer at Intrinsic.
+* interested in intersection between tooling and security (interesting challenges).
+
+## Introduction/Mission
+
+* *Chris*: if you had to write a bundler, or scaffolding tool, how could APIs
+  with Node.js make your life easier for you.
+  * ran user feedback session that indicated there was room for improvement
+    in some APIs.
+
+## Initial Successes
+
+* `mkdirp`:
+  * just landed in Node.js core.
+  * process of landing was a bit bumpy, due to questions around feature
+    detection.
+* `readdir`:
+  * you can now fetch stat information without crossing the JavaScript C++
+    boundary multiple times.
+* V8 Coverage:
+  * you can now dump coverage information directly from V8.
+  
+### What We Learned
+
+* *Ben*: we've had some good successes, but we learned there were some
+  shortcomings around feature detection -- maybe we need to formalize how we
+  propose features to work on ... brings us to.
+  
+## Backwards Compatibility and Polyfills
+
+* *Chris*:
+  * want feature detection in Node.js, but don't want to litigate it right now.
+  * for time being, perhaps we can get away with `process.version`.
+  * good start would be making pulls to community:
+    * update `mkdirp`, `rimraf`, based on `process.version`.
+    * _Michel_ also advocates this approach.
+
+* *Daniel*:
+  * how do we get adoption if only cutting edge Node.js should be using?
+  * perhaps backporting should be goal when possible?
+
+* *Bryan/Chris*:
+  * backporting is a pain in the butt, perhaps we just agree to polyfill;
+    see: readable stream.
+
+* *Ruy*:
+  * there are good aspects to not backporting features, drive more people
+    to adopt new Node.js versions.
+
+* *Ian*:
+  * because we're targetting CLI tools, perhaps we can be more aggressive
+    about pushing people to upgrade to new Node.js:
+    
+* *Daniel*:
+  * on topic of polyfills, `graceful-fs` is an example of 
+    _how not to polyfill_.
+
+* *Guy*:
+  * fs-extra is a better approach, doesn't mutate global fs object
+
+we seem to have reached relative consensus that:
+
+* we'll shim old library versions.
+* won't backport to old Node.js versions.
+* will use `process.version` for feature detecton.
+
+## What other sorts of features should we be working on?
+
+* *Chris*:
+  * argument parsing could be made better than `process.argv.slice(2)`.
+
+* *Bryan*:
+  * would like to see `execvp` implemented, this could be a good alternative
```

Comment:
Is it a concern that the Windows process model doesn't have anything like Unix exec, so this is likely to be unix only?

Reply:
cc @bengl ^

Reply:
Sam and I ended up discussing this in person. The short answer to Sam's question is: Yes.

We may be able to do something that behaves similarly enough for the use-cases we're aiming for.

Here's how things seem to have worked (or not) in python-land (thanks Sam for sending these to me!): https://stackoverflow.com/questions/7004687/os-exec-on-windows https://bugs.python.org/issue9148

So yes it might be the case that this needs to be a UNIX-only feature (there is some precedent for UNIX-only features in Node.js).

Reply:
I think some of the early feedback from the tooling user-feedback effort is that we want LESS platform specific behaviour versus more.

Reply:
@mhdawson Could something like `execvp` be implemented in a native module?  Is there a precedent for something like that, where Node.js provides a hook for native modules, but no corresponding JS API surface?  Is this a terrible idea?

Reply:
@boneskull Depending on the complexity you want, `execvp()` should be pretty easy to implement as a native module, even without any hooks from Node.js. You don’t have to worry about options like environment variables or working directories, since you can set any of these from the running process before jumping into `execvp()`, and IPC shouldn’t be a concern either.

Reply:
@bengl Would that ^ work for you?

Reply:
It already exists as several native module: https://www.npmjs.com/package/native-exec, https://www.npmjs.com/package/kexec, and probably more.

I'd rather see it in core so that userland native addons aren't necessary to get this done. I don't think process management functionality should be relegated to userland.